src/act-runner/runTestCases.ts

```diff
@@ -0,0 +1,56 @@
+import * as fs from 'fs'
+import * as path from 'path';
+import { Config, TestCase, Assertion, EarlReport } from '../types';
+import { groupedTestCases, loadTestCases } from './groupTestCases';
+import { runRuleTests, PageRunner } from './runRuleTests';
+
+export const runTestCases = async (
+  config: Config, 
+  pageRunner: PageRunner
+): Promise<EarlReport[]> => {
+  const { testCaseJson, ruleId } = config;
+  const cacheResults = getCacheResults(config);
+  const testCases = await loadTestCases(testCaseJson);
+  const tests = groupedTestCases(testCases, ruleId);
+  const results = [];
+
+  for (let testcases of tests) {
+    const { ruleName, ruleId } = testcases[0];
+    console.log(`Testing: ${ruleName} (${ruleId})`);
+    process.stdout.write("  ");
+
+    const untestedCases: TestCase[] = [];
+    testcases.forEach(testcase => {
+      const testcaseCache = cacheResults
+        .filter(({ subject }) => {
+          return subject.source.includes(testcase.relativePath);
+        })
+        .map(assertion => ({ "@graph": [assertion] }));
+
+      if (testcaseCache.length > 0) {
+        results.push(...testcaseCache);
+      } else {
+        untestedCases.push(testcase);
+      }
+    });
+    // Show cached results
+    process.stdout.write(",".repeat(testcases.length - untestedCases.length));
```

Comment:
So the amount of commas represent the number of untested cases? Why not output a human readable string with that information instead?

Reply:
Issue #3